### ISSUE TITLE
Question/feature req: Reject too big images

### ISSUE THREAD

**Issue by papandreou** [lovell/sharp#115]
In this app I'm building I need to be able to reject an input image if its dimensions are too big. Currently my code is divided up so there's a processing layer that sets up a "brigade" of proxy streams that the image is to be piped through, and now I need to enforce a pixel limit, sort of like http://www.graphicsmagick.org/GraphicsMagick.html#details-limit aka `gm convert -limit Pixels <number>`. What I've arrived at is roughly this:

```javascript
// inStream: Readable stream delivering the input image
// outStream: Writable stream that needs to receive the processed image
var sharp = require('sharp'),
    sharpInstance = sharp();
sharpInstance.metadata(function (err, metadata) {
    // if (err) ...
    if (metadata.width * metadata.height > areaLimit) {
        // Oh no, abandon ship...
        sharpInstance.emit('error', new Error('Pixel limit exceeded'));
    } else {
        sharpInstance.resize(...).pipe(outStream);
    }
})

inStream.pipe(sharpInstance);
```

... But that doesn't quite fit within the scheme I'm going for where each operation is encapsulated by a single proxy stream.

My first attempt ran `.pipe(outStream)` immediately, but that (perhaps predictably) caused the image to be read into memory in its entirety, then piped out.

What I would like to be able to is something like:

```javascript
var sharpInstance = sharp();
sharpInstance.metadata(function () {
    if (metadata.width * metadata.height > areaLimit) {
        sharpInstance.abort(); // Or something
    }
}).resize(...).pipe(outStream);
```

Or even:

```javascript
inStream.pipe(sharp()).limit(function (metadata) {
    return metadata.width * metadata.height > areaLimit;
}).resize(...).pipe(outStream);
```

... where specifying the limit as a number could be a shorthand.

What do you think?

**Comment by lovell** [comment]
@papandreou The `intake` branch now includes the [limitInputPixels](https://github.com/lovell/sharp/tree/intake#limitinputpixelspixels) option, which I think provides most of what you were originally looking for.

**Comment by lovell** [comment]
The `intake` branch has now been merged to master and will be released as v0.9.0 shortly.

**Comment by papandreou** [comment]
That sounds great. Actually I was just playing with it in my project while adding a bunch of new tests. Seems to work exactly as specified, so you have my blessing to go ahead with the release :)
